src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -576,6 +563,24 @@ def create_compile_context(self, target, target_workdir):
         sources=sources,
       ))
 
+  # TODO: This overrides the same method in jvm_compile.py, and that method should be refactored to
+  # allow fetching the digests for valid targets in a batch instead of one of a time.
+  def _classpath_for_context(self, context):
+    cp = super(RscCompile, self)._classpath_for_context(context)
+    assert isinstance(cp, ClasspathEntry)
+    if cp.directory_digest is not None:
+      return cp
+
+    cp_digest = Digest.load(os.path.dirname(cp.path))
```

Comment:
It's important to always actually call `capture_snapshots`, because it's possible that the `Store` no longer holds the `Digest`, in which case we need to re-snapshot to capture it.

So recommend to call like:
```
cp_globs = PathGlobsAndRoot(
  PathGlobs((fast_relpath(cp.path, get_buildroot()),)),
  text_type(get_buildroot(),
  Digest.load(os.path.dirname(cp.path)),
))
cp_snapshot, = self.context._scheduler.capture_snapshots((cp_globs,))
cp_digest = cp_snapshot.directory_digest
```

Also, the Digest needs to be stored explicitly:
```
cp_digest.dump(..)
```

See https://github.com/pantsbuild/pants/blob/d72fb81e0a2b8be81d22846047aefd480016dfa9/src/python/pants/backend/jvm/tasks/resolve_shared.py#L50-L59